kv/store_base.go

```diff
@@ -401,8 +407,12 @@ func (i *Iter) Next(ctx context.Context) (key []byte, val interface{}) {
 	case len(i.prefix) > 0:
 		i.seek(ctx)
 		nextFn = func() ([]byte, []byte) {
-			kv := <-i.seekChan
-			return kv.k, kv.v
+			select {
+			case <-ctx.Done():
+				return nil, nil
```

Comment:
not sure why does it return nil here

Reply:
good question, the `nil, nil` is a signal to the `Iter.isNext` call, it tells it the loop should exit